fetch/fetch.go

```diff
@@ -650,3 +756,90 @@ func (f *Fetch) GetHash(hash types.Hash32, h Hint, validateHash bool) chan HashD
 
 	return resChan
 }
+
+func (f *Fetch) mapPeerToHash(hash types.Hash32, peer p2p.Peer) {
```

Comment:
these functions are repetitive. please simply organize a big list of hashes on the caller side.
fetch just needs to have one interface like `RegisterPeerHashes(p2p.Peer, []types.Hash32)`

Reply:
done, thanks